python_modules/libraries/dagster-gcp/dagster_gcp/bigquery/io_manager.py

```diff
@@ -0,0 +1,239 @@
+        service account key as the "gcp_credentials" configuration. Dagster will store this key in a
+        temporary file and set GOOGLE_APPLICATION_CREDENTIALS to point to the file. After the run completes,
+        the file will be deleted, and GOOGLE_APPLICATION_CREDENTIALS will be unset.
+
+    """
+
+    @io_manager(
+        config_schema={
+            "dataset": Field(
+                StringSource,
+                description=(
+                    "Name of the BigQuery dataset to use. If not provided, the last prefix before"
+                    " the asset name will be used."
+                ),
+                is_required=False,
+            ),
+            "project": Field(StringSource, description="The GCP project to use."),
+            "location": Field(
+                Noneable(StringSource),
+                is_required=False,
+                default_value=None,
+                description="The GCP location.",
+            ),
+            "gcp_credentials": Field(
+                Noneable(StringSource),
+                is_required=False,
+                default_value=None,
+                description=(
+                    "GCP authentication credentials. If provided, a temporary file will be created"
+                    " with the credentials and GOOGLE_APPLICATION_CREDENTIALS will be set to the"
+                    " temporary file."
+                ),
+            ),
+        }
+    )
+    def bigquery_io_manager(init_context):
+        """I/O Manager for storing outputs in a BigQuery database
+
+        Assets will be stored in the dataset and table name specified by their AssetKey.
+        Subsequent materializations of an asset will overwrite previous materializations of that asset.
+        Op outputs will be stored in the dataset specified by output metadata (defaults to public) in a
+        table of the name of the output.
+
+        Note that the BigQuery config is mapped to the DB IO manager table hierarchy as follows:
+          BigQuery   DB IO
+        * project -> database
+        * dataset -> schema
+        * table -> table
+        """
+        mgr = DbIOManager(
+            type_handlers=type_handlers,
+            db_client=BigQueryClient(),
+            io_manager_name="BigQueryIOManager",
+            database=init_context.resource_config["project"],
+            schema=init_context.resource_config.get("dataset"),
+        )
+        if init_context.resource_config.get("gcp_credentials"):
+            if os.getenv("GOOGLE_APPLICATION_CREDENTIALS") is not None:
+                raise DagsterInvalidDefinitionError(
+                    "Resource config error: gcp_credentials config for BigQuery IO manager cannot"
+                    " be used if GOOGLE_APPLICATION_CREDENTIALS environment variable is set."
+                )
+            with tempfile.NamedTemporaryFile("w+", delete=False) as f:
```

Comment:
Something that strikes me as risky here is that the temporary file could get deleted after this block exits and before we yield.

Could we put the `yield` within the `with` block and use `f.flush` to make sure the contents get written out?

Reply:
yeah flush worked!